Kernel/FileSystem/SysFS/Subsystems/Kernel/PowerStateSwitch.cpp

```diff
@@ -45,28 +46,36 @@ ErrorOr<void> SysFSPowerStateSwitchNode::truncate(u64 size)
 
 ErrorOr<size_t> SysFSPowerStateSwitchNode::write_bytes(off_t offset, size_t count, UserOrKernelBuffer const& data, OpenFileDescription*)
 {
-    if (Checked<off_t>::addition_would_overflow(offset, count))
-        return EOVERFLOW;
-    if (offset > 0)
-        return EINVAL;
-    if (count > 1)
-        return EINVAL;
-
     char buf[1];
     TRY(data.read(buf, 1));
-    switch (buf[0]) {
-    case '0':
-        return EINVAL;
-    case '1':
-        reboot();
-        VERIFY_NOT_REACHED();
-    case '2':
-        poweroff();
-        VERIFY_NOT_REACHED();
-    default:
-        return EINVAL;
-    }
-    VERIFY_NOT_REACHED();
+    return Process::current().jail().with([&](auto& my_jail) -> ErrorOr<size_t> {
+        // Note: If we are in a jail, don't let the current process to change the power state.
+        if (my_jail)
+            return Error::from_errno(EPERM);
```

Comment:
Can we scope this spinlock to only the check for having a jail? 

Or is there some kind of race condition with starting a process that immediately tries to shutdown the system and then putting it in a jail? I thought that the current mechanism didn't allow for moving an existing process though. So if there's no such facility, it should be safe to keep everything after the "do I have a jail" check outside the jail spinlock.